### PR TITLE
Search feature test

### LINKED ISSUE #2276: Write feature tests for searching for Sinopia resources
Including:
* Filtering
* Sorting
* Paging
* When no results

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

add feature test coverage for various basic search result display behaviors.

connects with #2276

## How was this change tested?

PR only adds tests

## Which documentation and/or configurations were updated?

n/a

### CODE REVIEW

File: __tests__/feature/search.test.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    renderApp()
+
+    fireEvent.click(screen.getByText(/Linked Data Editor/, { selector: 'a' }))
+    fireEvent.click(screen.getByText(/Search/, { selector: 'a' }))
+    fireEvent.change(screen.getByLabelText('Query', { selector: 'input#searchInput' }), { target: { value: '*' } })
+    fireEvent.click(screen.getByLabelText('Submit search', { selector: 'button' }))
+
+    await screen.findByText(/foo bar/)
+
+    // TODO: why don't filtering options show up in test UI?
```
Comment (jmartin-sul) [inline __tests__/feature/search.test.js:157]: using `screen.debug()`, afaict, the filter elements were not rendering in the page for the tests.  i haven't figured out why, though i suspect maybe my fixture search results are incomplete?  but i didn't see anything obviously related missing from them, compared to what i saw when i ran searches on my local instance and looked at the response traffic in the browser's network debugger.

**Review by jmartin-sul — COMMENTED** [review]
i'm inclined to file a follow-on ticket for the TODO, and to get the rest merged, assuming the rest of the testing meets the other goals outlined in #2276.

File: __tests__/feature/search.test.js (hunk 1 above)
Comment (jermnelson) [reply __tests__/feature/search.test.js:157]: Created issue https://github.com/LD4P/sinopia_editor/issues/2499

**Review by jermnelson — APPROVED** [review]
Looks good. I found it interesting the use of both `screen.findByText` and the `findByText` function from the React testing library.

**Comment by jmartin-sul** [comment]
> I found it interesting the use of both `screen.findByText` and the `findByText` function from the React testing library.

ah, yeah, meant to comment on that specifically:  i was doing that just because it was the only reliable way i found to search for text within a given node that'd already been retrieved, and i wanted to retrieve the nodes in first column in one go using `screen.getAllByText`, so that i could check that the UI presented things  in the desired order.  but if there's a way to do that with `screen.findByText`, or some other more idiomatic approach, i'm happy to revisit.

**Pull request merged**